engine/gui/src/gui.cpp

```diff
@@ -1447,14 +1452,31 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
             c->m_StencilScopeIndices.SetCapacity(new_capacity);
         }
 
+        uint32_t num_pruned = 0;
         for (uint32_t i = 0; i < node_count; ++i)
         {
-            const RenderEntry& entry = c->m_RenderNodes[i];
+            RenderEntry& entry = c->m_RenderNodes[i];
             uint16_t index = entry.m_Node & 0xffff;
             InternalNode* n = &scene->m_Nodes[index];
             float opacity = 1.0f;
+
+            // Ideally, we'd like to be able to do this in an Update step,
+            // but since the gui script is updated _after_ the gui component, we need to accomodate
+            // for any late scripting changes
+            // Note: We need this update step for bones as well, since they update their transform
```

Comment:
Sadly, this is actually the update for the rendered nodes, and we need this for the gui bones to end up in the correct place (see extension-spine example scene where we attach objects to the gui bones)